mongo/errors.go

```diff
@@ -122,25 +118,28 @@ func (e CommandError) HasErrorLabel(label string) bool {
 	return false
 }
 
-// IsMaxTimeMSExpiredError indicates if the error is a MaxTimeMSExpiredError.
+// IsMaxTimeMSExpiredError returns true if the error is a MaxTimeMSExpired error.
 func (e CommandError) IsMaxTimeMSExpiredError() bool {
 	return e.Code == 50 || e.Name == "MaxTimeMSExpired"
 }
 
-// WriteError is a non-write concern failure that occurred as a result of a write
-// operation.
+// WriteError is an error that occurred during execution of a write operation. This error type is only returned as part
+// of a WriteException or BulkWriteException.
 type WriteError struct {
-	Index   int
+	// The index of the write in the slice passed to an InsertMany or BulkWrite operation that caused this error.
+	// See the Collection.InsertMany documentation for an example of how this can be used.
```

Comment:
Are you gong to add an example in this PR?  If not, best to take this out until you do.

Reply:
Removed for now. I'd prefer to move examples for the different types of errors to a different PR.